lib/matplotlib/widgets.py

```diff
@@ -809,6 +809,24 @@ def on_changed(self, func):
         """
         return self._observers.connect('changed', lambda val: func(val))
 
+    def update_range(self, vmin=None, vmax=None):
+        """Update the range of the slider"""
+        if not vmin and not vmax:
+            raise ValueError(
+                (f"Argument vmin ({type(vmin)}) has no value"
+                    f"Argument vmax ({type(vmax)}) has no value"))
+        if vmin:
+            self.valmin = vmin
+        if vmax:
+            self.valmax = vmax
+        self.val = self._value_in_bounds(self.valinit)
```

Comment:
I think this should move the current value, not the initial value.

Reply:
I have done the required change